ext/vulnsrc/redhat/redhat.go

```diff
@@ -252,19 +271,76 @@ func CollectVulnsForAdvisory(advisoryDefinition ParsedAdvisory, ovalDoc OvalV2Do
 				}
 
 			}
+			log.WithFields(log.Fields{
+				"name":     vulnerability.Name,
+				"packages": len(advisoryDefinition.PackageList),
+				"affected": len(vulnerability.Affected),
+			}).Trace("Append vulnerability")
 			if len(vulnerability.Affected) > 0 {
-				vulnerabilities = append(vulnerabilities, vulnerability)
+				// check the pending vulnerabilities set for this vulnerability (since they can appear in multiple manifest entries)
+				if pendingVulnNames[cve.Value+" - "+ParseRhsaName(advisoryDefinition)] {
+					// this vuln has already been added to the set, so skip so we don't end up with duplicates
+					log.Trace(fmt.Sprintf("Filtering unique package info for already-queued vulnerability: %s",
+						cve.Value+" - "+ParseRhsaName(advisoryDefinition)))
+					// get the slice index for the existing copy of this vuln, so we can modify it instead of duplicating it
+					i := GetPendingVulnerabilitySliceIndex(vulnerabilities, vulnerability)
+					if i >= 0 {
+						// merge any new unique vuln features into the existing vuln
+						vulnerabilities[i] = MergeVulnerabilityFeature(vulnerability, vulnerabilities[i])
```

Comment:
`CollectVulnsForAdvisory` takes a single oval file and return vulnerabilities.. how this is going to work when the same combination of RHSA + CVE is also present in different oval file? Can you do a merging outside of `CollectVulnsForAdvisory` when all data are collected?